src/argilla/client/feedback/dataset/remote/dataset.py

```diff
@@ -141,27 +250,187 @@ def __init__(
+            questions: contains the questions that will be used to annotate the dataset.
+            metadata_properties: contains the metadata properties that will be indexed
+                and could be used to filter the dataset. Defaults to `None`.
+            guidelines: contains the guidelines for annotating the dataset. Defaults to `None`.
+
+        Raises:
+            TypeError: if `fields` is not a list of `FieldSchema`.
+            ValueError: if `fields` does not contain at least one required field.
+            TypeError: if `questions` is not a list of `TextQuestion`, `RatingQuestion`,
+                `LabelQuestion`, and/or `MultiLabelQuestion`.
+            ValueError: if `questions` does not contain at least one required question.
+            TypeError: if `guidelines` is not None and not a string.
+            ValueError: if `guidelines` is an empty string.
+        """
+
+        self._fields = fields
+        self._fields_schema = None
+        self._questions = questions
+        self._metadata_properties = metadata_properties
+        self._guidelines = guidelines
+        # TODO: uncomment once supported by the API
+        # self._allow_extra_metadata = allow_extra_metadata
+
+        self._client = client  # Required to be able to use `allowed_for_roles` decorator
+        self._id = id
+        self._name = name
+        self._workspace = workspace
+        self._created_at = created_at
+        self._updated_at = updated_at
+
+        self._records = RemoteFeedbackRecords(dataset=self)
+
+    @property
+    def records(self) -> RemoteFeedbackRecords:
+        """Returns an instance of `RemoteFeedbackRecords` that allows you to iterate over
+        the records in the dataset. The records are fetched from Argilla on the fly and
+        not stored in memory. You can also iterate over the records directly from the
+        dataset instance.
+        """
+        return self._records
+
+    @property
+    def id(self) -> "UUID":
+        """Returns the ID of the dataset in Argilla."""
+        return self._id
+
+    @property
+    def name(self) -> str:
+        """Returns the name of the dataset in Argilla."""
+        return self._name
+
+    @property
+    def workspace(self) -> "Workspace":
+        """Returns the workspace the dataset belongs to in Argilla."""
+        return self._workspace
+
+    @property
+    def url(self) -> str:
+        """Returns the URL of the dataset in Argilla."""
+        return f"{self._client.base_url}/dataset/{self.id}/annotation-mode"
+
+    @property
+    def created_at(self) -> datetime:
+        """Returns the datetime when the dataset was created in Argilla."""
+        return self._created_at
+
+    @property
+    def updated_at(self) -> datetime:
+        """Returns the datetime when the dataset was last updated in Argilla."""
+        return self._updated_at
+
+    def __repr__(self) -> str:
```

Comment:
maybe this can be formatted a bit more nicely with multi_line. Also, I would say thing like the filters and sorts should be added, when present.

Reply:
Agree, and I think this is a good first issue for someone to contribute for